controllers/chat/moveToAnon.js

```diff
@@ -40,10 +46,26 @@ const moveToAnon = async (req, res) => {
   const oldChannelDetail = await client.queryChannels(filter, sort);
   //req.userId always anon since the token are using anon token
   const mySignedId = await UsersFunction.findSignedUserId(User, req.userId);
-  const prevTargetUser = Object.values(oldChannelDetail[0].data.better_channel_member).filter(
-    (user) => user.user_id !== mySignedId
-  )[0];
-  const detailPrevTargetUser = await UsersFunction.findUserById(User, prevTargetUser.user_id);
+
+  let prevTargetUser;
+  if (oldChannelDetail[0].data.better_channel_member) {
+    prevTargetUser = Object.values(oldChannelDetail[0].data.better_channel_member).filter(
+      (user) => user.user_id !== mySignedId
+    )[0];
+    targetUserId = prevTargetUser.user_id;
+  }
+
+  if (!prevTargetUser) {
+    prevTargetUser = {
+      user_id: targetUserId,
+      anon_user_info_color_code: color.code,
+      anon_user_info_color_name: color.color,
+      anon_user_info_emoji_code: emoji.emoji,
+      anon_user_info_emoji_name: emoji.name
+    };
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for identifying the previous target user has been altered. The new logic checks if `better_channel_member` data exists in the old channel detail. If it does, it filters out the current user and assigns the previous target user. If not, it creates a new object with the target user's information. This seems like a reasonable change, but ensure that this new logic aligns with your application's requirements.
```diff
+   let prevTargetUser;
+   if (oldChannelDetail[0].data.better_channel_member) {
+     prevTargetUser = Object.values(oldChannelDetail[0].data.better_channel_member).filter(
+       (user) => user.user_id !== mySignedId
+     )[0];
+     targetUserId = prevTargetUser.user_id;
+   }
+ 
+   if (!prevTargetUser) {
+     prevTargetUser = {
+       user_id: targetUserId,
+       anon_user_info_color_code: color.code,
+       anon_user_info_color_name: color.color,
+       anon_user_info_emoji_code: emoji.emoji,
+       anon_user_info_emoji_name: emoji.name
+     };
+   }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->